0000-deb-repo-additional-headers.md

```diff
@@ -88,6 +88,50 @@ The integrity of the `deb` binary packages must be verified using the checksums
 
 This will be handled in a separate RFC.
 
+## Failure Registry
+
+Packages might have sync issues, such as wrong GPG signature or
+checksum mismatch or downloading issues etc. However, broken packages
+still can be assigned to the channels/repos and thus potentially
+invaliadate them. In some circumstances this might have serious impact
+on the infrastructure update performance. To avoid that, we need to
+have a mechanism that will verify if a particular channel is "clean",
+i.e. no any broken packages involved.
+
+### Use Case Scenario Example
+
+Following example should be able to happen:
+
+1. An admin runs reposync in a background.
+2. Broken package appears, reposync is writing an error in the log, as
+   usual.
+3. The error is not noticed by another admin on a shift (UX is unused
+   at the moment, other circumstances).
+4. Synchronised channel is noticed and thus scheduled for an update.
+5. An update cannot be completed due to broken package.
+
+In this case an admin did not noticed that the channel contains broken
+packages and thus should not be recommended for an update, but needs
+to be fixed first.
+
+### Solution
+
+The solution is simply add the following table:
+
+```
+rhnPackageErrors
+================
+package_id  NUMERIC NOT NULL
+error       VARCHAR NOT NULL
+```
```

Comment:
We could also go for "channel errors" and not for individual packages. Need some investigation what is easier/better to implement and review by admins when they get the errors.

Reply:
We currently send one UI notification per channel on successful or erroneous `repo-sync`. Now if I understand correctly, on download or verification errors of single packages the `repo-sync` process would still complete successfully for this channel? In this case I guess there could be a third state like "`repo-sync` completed with errors", but why don't we just call it failed and discard the whole attempt?

Reply:
Thinking again about this I have to say that this cannot work. If a package has a problem during download and the checksum is not the expected one, the package is not imported at all. 
This means there is no package ID where we can store the error.

I would go for a channel error and compile a text with all failed downloads (packages and patches)